test/media/drm_engine_unit.js

```diff
@@ -1093,6 +1093,178 @@ describe('DrmEngine', () => {
           shaka.util.Error.Code.FAILED_TO_GENERATE_LICENSE_REQUEST,
           message, nativeError, undefined));
     });
+
+    it('should throw a OFFLINE_SESSION_REMOVED error', async () => {
+      // Given persistent session is not available
+      session1.load.and.returnValue(false);
+
+      onErrorSpy.and.stub();
+
+      await drmEngine.initForPlayback(
+          manifest.variants, ['persistent-session-id']);
+      await drmEngine.attach(mockVideo);
+
+      expect(drmEngine.initialized()).toBe(true);
+
+      await Util.shortDelay();
+
+      expect(mockMediaKeys.createSession).toHaveBeenCalledTimes(1);
+      expect(mockMediaKeys.createSession)
+          .toHaveBeenCalledWith('persistent-license');
+      expect(session1.load).toHaveBeenCalledWith('persistent-session-id');
+
+      expect(onErrorSpy).toHaveBeenCalled();
+      const error = onErrorSpy.calls.argsFor(0)[0];
+      shaka.test.Util.expectToEqualError(error, new shaka.util.Error(
+          shaka.util.Error.Severity.CRITICAL,
+          shaka.util.Error.Category.DRM,
+          shaka.util.Error.Code.OFFLINE_SESSION_REMOVED));
+    });
+
+    it('uses persistent session ids when available', async () => {
+      const Uint8ArrayUtils = shaka.util.Uint8ArrayUtils;
+
+      const keyId1 = makeKeyId(1);
+      const keyId2 = makeKeyId(2);
+
+      /** @type {!Uint8Array} */
+      const initData1 = new Uint8Array(5);
+
+      // Key IDs in manifest
+      tweakDrmInfos((drmInfos) => {
+        drmInfos[0].keyIds = new Set([
+          Uint8ArrayUtils.toHex(keyId1), Uint8ArrayUtils.toHex(keyId2),
+        ]);
+        drmInfos[0].initData = [
+          {initData: initData1, initDataType: 'cenc', keyId: null},
+        ];
+      });
+
+      // Given persistent session is available
+      session1.load.and.returnValue(true);
+
+      config.persistentSessionOnlinePlayback = true;
+      config.persistentSessionsMetadata = [{
+        sessionId: 'persistent-session-id',
+        initData: initData1,
+        initDataType: 'cenc'}];
+
+      drmEngine.configure(config);
+
+      await initAndAttach();
+
+      await Util.shortDelay();
+
+      session1.keyStatuses.forEach.and.callFake((callback) => {
+        callback(keyId1, 'usable');
+        callback(keyId2, 'usable');
+      });
+
+      session1.on['keystatuseschange']({target: session1});
+
+      expect(mockMediaKeys.createSession).toHaveBeenCalledTimes(1);
+      expect(mockMediaKeys.createSession)
+          .toHaveBeenCalledWith('persistent-license');
+      expect(session1.load).toHaveBeenCalledWith('persistent-session-id');
+
+      expect(session2.generateRequest).not.toHaveBeenCalled();
+    });
+
+    it(
```

Comment:
I think this would fit into one line. No need to do the weird indenting.

Reply:
The whole line
```
it('tries persistent session ids before requesting a license', async () => {
```
is 81 characters long 😞